Source/Dafny/Resolver.cs

```diff
@@ -3320,145 +3323,144 @@ public class ModuleBindings
               }
               if ((BigInteger)e1 >= bin.Type.AsBitVectorType.Width) {
                 return BigInteger.Zero;
               }
               return ((BigInteger)e0 << (int)(BigInteger)e1) & MaxBV(bin.E0.Type);
             }
-            case BinaryExpr.Opcode.RightShift: {
+            case BinaryExpr.ResolvedOpcode.RightShift: {
               if ((BigInteger)e1 < 0) {
-                reporter.Error(MessageSource.Resolver, bin.E1,
-                  "Shift amount may not be negative: " + ((BigInteger)e1).ToString());
-                return null;
+                return null; // Negative shift
               }
               if ((BigInteger)e1 >= bin.Type.AsBitVectorType.Width) {
                 return BigInteger.Zero;
               }
               return (BigInteger)e0 >> (int)(BigInteger)e1;
             }
-            case BinaryExpr.Opcode.And: return (bool) e0 && (bool) e1;
-            case BinaryExpr.Opcode.Or: return (bool) e0 || (bool) e1;
-            case BinaryExpr.Opcode.Iff: return (bool) e0 == (bool) e1; // <==>
-            case BinaryExpr.Opcode.Imp: return !(bool) e0 || (bool) e1; // ==>
-            case BinaryExpr.Opcode.Exp: return !(bool) e0 || (bool) e1; // <==, arguments reversed during parse
-            case BinaryExpr.Opcode.Gt:
+            case BinaryExpr.ResolvedOpcode.And: return (bool) e0 && (bool) e1;
+            case BinaryExpr.ResolvedOpcode.Or: return (bool) e0 || (bool) e1;
+            case BinaryExpr.ResolvedOpcode.Iff: return (bool) e0 == (bool) e1; // <==>
+            case BinaryExpr.ResolvedOpcode.Imp: return !(bool) e0 || (bool) e1; // ==> and <==
+            case BinaryExpr.ResolvedOpcode.Gt:
               if (isInt) return (BigInteger) e0 > (BigInteger) e1;
               if (isBV) return (BigInteger) e0 > (BigInteger) e1;
               if (isReal) return (Basetypes.BigDec) e0 > (Basetypes.BigDec) e1;
+              break;
+            case BinaryExpr.ResolvedOpcode.GtChar:
               if (bin.E0.Type.IsCharType) return ((string) e0)[0] > ((string) e1)[0];
               break;
-            case BinaryExpr.Opcode.Ge:
+            case BinaryExpr.ResolvedOpcode.Ge:
               if (isInt) return (BigInteger) e0 >= (BigInteger) e1;
               if (isBV) return (BigInteger) e0 >= (BigInteger) e1;
               if (isReal) return (Basetypes.BigDec) e0 >= (Basetypes.BigDec) e1;
+              break;
+            case BinaryExpr.ResolvedOpcode.GeChar:
               if (bin.E0.Type.IsCharType) return ((string) e0)[0] >= ((string) e1)[0];
               break;
-            case BinaryExpr.Opcode.Lt:
+            case BinaryExpr.ResolvedOpcode.Lt:
               if (isInt) return (BigInteger) e0 < (BigInteger) e1;
               if (isBV) return (BigInteger) e0 < (BigInteger) e1;
               if (isReal) return (Basetypes.BigDec) e0 < (Basetypes.BigDec) e1;
               if (bin.E0.Type.IsCharType) return ((string) e0)[0] < ((string) e1)[0];
```

Comment:
I think this line should be deleted (if it's a `char`, isn't the operator an `LtChar` and this case handled on L3447?)

Reply:
Yes - missed deleting that line.